src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelTest.java

```diff
@@ -817,27 +835,52 @@ public void testBigAvroBufferBytesThreshold() throws Exception {
   }
 
   @Test
-  public void testRegisterJmxMetrics() {
-    MetricRegistry metricRegistry = Mockito.spy(MetricRegistry.class);
-    MetricsJmxReporter metricsJmxReporter =
-        Mockito.spy(new MetricsJmxReporter(metricRegistry, TEST_CONNECTOR_NAME));
+  public void testTopicPartitionChannelInvalidJmxReporter() throws Exception {
```

Comment:
lets also add a test where there are >1 channels and we verify the metrics values after inserting few records in each channel.

Reply:
check out `SnowflakeTelemetryPipeStatusMetricsIT`

Reply:
thanks for the suggestion, there's a similar test in the next PR but it only has one channel. 

Added metrics verification to testStreamingIngest_multipleChannelPartitions_withMetrics in SnowflakeSinkServiceV2IT.java